api/src/DataPersister/PeriodDataPersister.php

```diff
@@ -19,9 +22,56 @@ public function __construct(
         );
     }
 
+    public function beforeCreate($data): BaseEntity {
+        static::managePeriodDays($data);
+
+        return $data;
+    }
+
+    public function beforeUpdate($data): BaseEntity {
+        $orig = $this->em->getUnitOfWork()->getOriginalEntityData($data);
```

Comment:
Orig-Daten von Period bestimmen - ist das der korrekte Weg? 
Oder gibt es da eine bessere Alternative?

Reply:
Alternativ könntest du dir per DI `public RequestStack $requestStack` injizieren lassen und dann mit `$this->requestStack->getCurrentRequest()->attributes->get('previous_data')` die vorherigen Daten rausholen. Das ist aber vor allem nützlich wenn man auch die vormaligen Relations braucht. Und ausserdem würde das dann das vormalige Camp oder die vormalige Periode geben, je nach dem ob ich gerade auf `/camps/1a2b3c4d` oder `/periods/1a2b3c4d` gepatcht habe (erlauben wir aktuell nicht, aber könnte ja in Zukunft mal sein).

Darum gefällts mir besser so wie dus jetzt gelöst hast.

Reply:
thx